mantidimaging/gui/windows/spectrum_viewer/spectrum_widget.py

```diff
@@ -55,19 +70,43 @@ def get_tof_range(self) -> tuple[int, int]:
         r_min, r_max = self.range_control.getRegion()
         return int(r_min), int(r_max)
 
-    def add_roi(self, roi: SensibleROI):
-        self.roi.setPos((roi.left, roi.top))
-        self.roi.setSize((roi.width, roi.height))
-        self.roi.maxBounds = self.roi.parentBounds()
-        self.roi.addScaleHandle([1, 1], [0, 0])
-        self.roi.addScaleHandle([1, 0], [0, 1])
-        self.roi.addScaleHandle([0, 0], [1, 1])
-        self.roi.addScaleHandle([0, 1], [1, 0])
-        self.image.vb.addItem(self.roi)
-
-    def get_roi(self) -> SensibleROI:
-        pos = CloseEnoughPoint(self.roi.pos())
-        size = CloseEnoughPoint(self.roi.size())
+    def add_roi(self, roi: SensibleROI, name: str = None) -> None:
+        """
+        Add an ROI to the image view.
+
+        :param roi: The ROI to add.
+        :param name: The name of the ROI.
+        """
+
+        my_roi = ROI(pos=(0, 0), rotatable=False, scaleSnap=True, translateSnap=True)
+
+        my_roi.setPos((roi.left, roi.top))
+        my_roi.setSize((roi.width, roi.height))
+        self.set_max_roi_size(roi.width, roi.height)
+        my_roi.maxBounds = my_roi.parentBounds()
+        my_roi.addScaleHandle([1, 1], [0, 0])
+        my_roi.addScaleHandle([1, 0], [0, 1])
+        my_roi.addScaleHandle([0, 0], [1, 1])
+        my_roi.addScaleHandle([0, 1], [1, 0])
+
+        self.roi_dict[name] = my_roi
+        self.roi_dict[name].sigRegionChanged.connect(self.roi_changed.emit)  # might not need this
+        self.image.vb.addItem(self.roi_dict[name])
+
+    def get_roi(self, roi_name: str = None) -> SensibleROI:
+        """
+        Get the ROI with the given name. If no name is given, the default ROI is returned.
+
+        :param roi_name: The name of the ROI to return.
+        :return: The ROI with the given name.
+        """
+        if roi_name in self.roi_dict.keys():
+            pos = CloseEnoughPoint(self.roi_dict[roi_name].pos())
+            size = CloseEnoughPoint(self.roi_dict[roi_name].size())
+            return SensibleROI.from_points(pos, size)
+
+        pos = CloseEnoughPoint((0, 0))
+        size = CloseEnoughPoint((self.max_roi_size[0], self.max_roi_size[1]))
```

Comment:
Mypy gives me `Argument 1 to "CloseEnoughPoint" has incompatible type "Tuple[int, int]"; expected "Union[List[int], List[float]]"`. Looking at the `__init__` in mantidimaging/core/utility/close_enough_point.py, it is the type hint there that needs updatinging. It should be
`Union[Sequence[int], Sequence[float]]`
with Sequence from
`from collections.abc import Sequence`
So that it would support anything subscriptable.

Reply:
Thanks for noticing this one, didn't seem to come up for me when initially committing as part of pre-commit checks. Made the suggested change to the type hint in core utils 👍